packages/data-point/lib/reducer-path/factory.js

```diff
@@ -13,27 +13,78 @@ const REDUCER_PATH = (module.exports.type = 'ReducerPath')
 function ReducerPath () {
   this.type = REDUCER_PATH
   this.name = ''
+  this.body = undefined
   this.asCollection = false
 }
 
 module.exports.ReducerPath = ReducerPath
 
+/**
+ * @param {*} source
+ * @returns {boolean}
+ */
 function isPath (source) {
   return _.isString(source) && source.charAt(0) === '$'
 }
 
 module.exports.isPath = isPath
 
+const getAccValue = acc => acc.value
+
+module.exports.getAccValue = getAccValue
+
+const getFromAcc = jsonPath => acc => _.get(acc, jsonPath)
+
+module.exports.getFromAcc = getFromAcc
+
+const getFromAccValue = jsonPath => acc => _.get(acc.value, jsonPath)
+
+module.exports.getFromAccValue = getFromAccValue
+
+const mapFromAccValue = jsonPath => acc => {
+  if (Array.isArray(acc.value)) {
+    return _.map(acc.value, jsonPath)
+  }
+
+  return null
+}
+
+module.exports.mapFromAccValue = mapFromAccValue
+
+/**
+ * @param {string} jsonPath
+ * @param {boolean} asCollection
+ * @returns {Function}
+ */
+function getPathReducerFunction (jsonPath, asCollection) {
+  if (jsonPath === '.' || _.isEmpty(jsonPath)) {
+    return getAccValue
+  }
+
+  if (jsonPath.startsWith('..')) {
+    return getFromAcc(jsonPath.slice(2))
+  }
+
+  if (asCollection) {
+    return mapFromAccValue(jsonPath)
+  }
+
+  return getFromAccValue(jsonPath)
+}
+
+module.exports.getPathReducerFunction = getPathReducerFunction
+
 /**
  * parse reducer
- * @param  {string} reducerRaw raw reducer path
+ * @param {string} source - raw reducer path
  * @return {reducer}
  */
 function create (source) {
   const reducer = new ReducerPath()
 
   reducer.asCollection = source.slice(-2) === '[]'
   reducer.name = _.defaultTo(source.substr(1), '.').replace(/\[]$/, '')
+  reducer.body = getPathReducerFunction(reducer.name, reducer.asCollection)
```

Comment:
thinking about the future of compiled reducers, and this name `body` doesn't sound right, thinking on `resolver` might be a better fit? any other ideas?

the reason I had used `body`, is because there was the string function and the non-string function since that is now removed, I think we can find a better name for it

Reply:
I agree, I think just `resolve` would make more sense. For now I think I should just leave it as body so it matches the naming in `ReducerFunction`

Reply:
+1 to `resolver`, ... I believe `body` was from function body, and I would almost bet, that it got carried to this module because of a copy paste 😄